<!doctype html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Technical bias</title>
  <meta name="description" content="">
  <meta name="author" content="Michael Love">
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
  <link rel="stylesheet" href="libraries/frameworks/revealjs/css/reveal.min.css">
  <link rel="stylesheet" href="libraries/frameworks/revealjs/css/theme/solarized.css" id="theme">
  <link rel="stylesheet" href="libraries/highlighters/highlight.js/css/tomorrow.css" id="theme">
  <!--[if lt IE 9]>
  <script src="lib/js/html5shiv.js"></script>
  <![endif]-->  
</head>
<body>
  <div class="reveal">
    <div class="slides">
      <section class='' data-state='' id='slide-1'>
  <h2>Technical bias in sequencing data</h2>
  <p>Michael Love</p>

<p>Research Fellow</p>

<p>Irizarry group, DFCI/HSPH</p>

<p>later: <a href="http://mikelove.github.io/techbias/">http://mikelove.github.io/techbias/</a></p>

</section>
<section>
   <section class='' data-state=''>
    <h2>Bias</h2>
    <p><img src="figure/ruler.jpg"></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <p><img src="figure/pothole.jpg"></p>

<p>thought experiment: measure potholes in Boston vs Cambridge</p>

<ol>
<li>everyone&#39;s rulers are off by +1 cm</li>
<li>Boston rulers left in the sun, stretched by +1 cm</li>
</ol>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <ol>
<li>bias cancels out</li>
<li>bias is correlated with comparison of interest</li>
</ol>

<p><br><br></p>

<p class='fragment'>...sounds simple, but still we see datasets with 100%
confounding of condition with experimental batch</p>

    <aside class='notes'>
      
    </aside>
   </section>
</section>
<section class='' data-state='' id='slide-3'>
  <h2>&quot;Sequencing bias&quot;</h2>
  <ul>
<li>this talk: bias in use of sequencing as a <em>quantitative</em> assay</li>
<li>there is also bias in sequencing, e.g. calling bases A,C,G,T</li>
</ul>

</section>
<section>
   <section class='' data-state=''>
    <h2>Example 1: DNA sequencing</h2>
    <ul>
<li>we are sequencing DNA for genotyping</li>
<li>meanwhile, use data to find copy number</li>
</ul>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>Copy number</h2>

<p><img src="figure/cnv_diagram.jpg"></p>

<p>relative to a reference genome</p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <p><img height=600 src="figure/illumina0.png"></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <p><img height=600 src="figure/illumina1.png"></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <p><img height=600 src="figure/illumina2.png"></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <p><img height=600 src="figure/illumina3.png"></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>Reads stored in &quot;Fastq&quot; file</h2>

<p><img src="figure/fastq.jpg"></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>Align reads to reference genome</h2>

<p><img src="figure/tview.png"></p>

<p>the local number of reads: &quot;read depth&quot;</p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>Local read depth</h2>

<p>changes in read depth relative to a reference:</p>

<p><img src="figure/yoon.png"></p>

<p><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3378858">Yoon et al 2009</a></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>Copy number</h2>

<p><img src="figure/cnv_diagram.jpg"></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>Bias</h2>

<p>deviation of coverage from that expected <br>
from proporitions of molecules in the &quot;pool&quot;</p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <p><img src="assets/fig/unnamed-chunk-1-1.png" alt="plot of chunk unnamed-chunk-1"> </p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <p><img src="assets/fig/unnamed-chunk-2-1.png" alt="plot of chunk unnamed-chunk-2"> </p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <p><img src="assets/fig/unnamed-chunk-3-1.png" alt="plot of chunk unnamed-chunk-3"> </p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <p><img src="assets/fig/unnamed-chunk-4-1.png" alt="plot of chunk unnamed-chunk-4"> </p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>Assumptions</h2>

<ul>
<li>uniform rate of coverage</li>
<li>or at least, sample / reference is ok</li>
</ul>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>Amplification</h2>

<p>involves polymerase copying DNA many times over</p>

<p><a href="http://www.rcsb.org/pdb/101/motm.do?momID=40">
<img src="figure/polymerase.jpg">
</a></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>Amplification</h2>

<ul>
<li>polymerase has a preference for certain number of C,G</li>
<li>slightly from different experiment to experiment</li>
</ul>

    <aside class='notes'>
      
    </aside>
   </section>
</section>
<section>
   <section class='' data-state=''>
    <h2>Observing GC bias</h2>
    <ol>
<li>partition the genome into windows</li>
<li>count the number of reads</li>
<li>calculate the ratio (C+G)/(A+C+G+T)</li>
</ol>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>Differential GC bias</h2>

<p><img src="figure/boeva.jpg"></p>

<p><a href="http://bioinformatics.oxfordjournals.org/content/27/2/268.long">Boeva et al 2010</a></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <p><img src="figure/benjamini1.png"></p>

<p><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3378858">Benjamini and Speed 2012</a></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <p><img src="figure/benjamini2.png"></p>

<p><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3378858">Benjamini and Speed 2012</a></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>Correction</h2>

<p><br></p>

<ul>
<li>divide counts by prediction from model, \(\hat{\mu}\)</li>
<li class='fragment'>then look at ratio of corrected counts</li>
<li class='fragment'>alternatively, put \(\hat{\mu}\) in the model</li>
</ul>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>Summary 1</h2>

<ul>
<li>GC bias was <em>different</em> between sample and reference</li>
<li class='fragment'>bias didn&#39;t cancel</li>
<li class='fragment'>modeling on features we generated <em>in silico</em> helped</li>
</ul>

    <aside class='notes'>
      
    </aside>
   </section>
</section>
<section>
   <section class='' data-state=''>
    <h2>Example 2: RNA sequencing</h2>
    <ul>
<li>we want to quantify mRNA and compare across patients</li>
<li>needed for research and as a marker in diagnostics</li>
</ul>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>RNA sequencing protocol</h2>

<p><a href="http://www.sciencedirect.com/science/article/pii/S0014482714000160">
<img src="figure/vandijk.jpg">
</a></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>Bias in RNA sequencing</h2>

<p>deviation of coverage from expected <br>
given the proportion of molecules in the pool</p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>a few sources of bias</h2>

<ul>
<li>fragmentation and size selection</li>
<li>primer ligation</li>
<li>PCR</li>
</ul>

<p><br></p>

<p class='fragment'>(other steps are certainly also important)</p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>fragmentation, size selection</h2>

<p><img height=500 src="figure/fragsize.png"></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>fragmentation, size selection</h2>

<p><img src="assets/fig/unnamed-chunk-5-1.png" alt="plot of chunk unnamed-chunk-5"> </p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>primer ligation</h2>

<p><img src="figure/hansenpriming.png"></p>

<ul>
<li>random hexamer priming biases read starts</li>
<li>weight each read using the observed freq&#39;s</li>
</ul>

<p><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2896536/">Hansen et al 2010</a></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <p><img src="figure/evans.jpg"></p>

<p>useful plot for identifying non-uniform coverage</p>

<p><a href="http://www.biomedcentral.com/1471-2105/11/430/">Evans, Hower and Pachter 2010</a></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <p><img src="figure/nonuniform.jpg"></p>

<p><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2898062/">Li, Jiang and Wong 2010</a></p>

<p>linear model of the Poisson rate including sequence bias</p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>Multiple isoforms per gene</h2>

<p><img src="figure/salzman.jpg"></p>

<p><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3846358/">Salzman, Jiang and Wong 2011</a></p>

<p>model for estimating isoform abundances including fragmentation, size
selection, sequence bias</p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <p>Probability of a vector of read counts \(\vec{n}\), indexed by read type
<em>j</em>:</p>

<p><br></p>

<p>\[ f_\theta(\vec{n}) = \prod_j f_{Pois}(n_j, \vec{\theta} \cdot \vec{a}_j ) \]</p>

<ul>
<li>\(\vec{\theta}\) are the isoform abundances</li>
<li>\(\vec{a}_j\) are the rates, 0 if read \(j\) cannot be generated by the
isoform</li>
<li>rates include bias</li>
</ul>

<p><br></p>

<p><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3846358/">Salzman, Jiang and Wong 2011</a></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>Multiple isoforms per gene</h2>

<p><img height=400 src="figure/roberts.jpg"></p>

<p><a href="http://genomebiology.com/2011/12/3/R22">Roberts et al 2011</a></p>

<p>likelihood of isoform abundances given fragment length
distribution and sequence bias: used in <em>Cufflinks</em></p>

    <aside class='notes'>
      
    </aside>
   </section>
</section>
<section>
   <section class='' data-state=''>
    <h2>Aggregate bias at the gene level</h2>
    <ul>
<li>model gene counts on known covariates</li>
<li>factor analysis</li>
<li>add batch to model formula</li>
</ul>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h3>conditional quantile normalization</h3>

<p><img height=500 src="figure/cqn.jpg"></p>

<p><a href="http://biostatistics.oxfordjournals.org/content/13/2/204.full">Hansen, Irizarry and Wu 2011</a></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h3>Bias in batches</h3>

<p><img src="figure/bottomly_batch.png"></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h3>Surrogate variable analysis</h3>

<p><img height=500 src="figure/sva.png"></p>

<p>SVA: <a href="http://www.plosgenetics.org/article/info%3Adoi%2F10.1371%2Fjournal.pgen.0030161">Leek et al 2007</a>,
svaseq: <a href="http://nar.oxfordjournals.org/content/early/2014/10/07/nar.gku864.abstract">Leek 2014</a></p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>Add batch to the model</h2>

<p><br></p>

<p>Per gene, model the mean for sample j, \(\mu_j\), as:</p>

<p><br></p>

<p>\[ log(\mu_j) = \beta_0 + \beta_{b} 1_{j \in B} + \beta_{t} 1_{j \in T} \]</p>

<p><br></p>

<p>where <em>B</em> is the second batch, <em>T</em> is the treated samples.</p>

    <aside class='notes'>
      
    </aside>
   </section>
   <section class='' data-state=''>
    <h2>Summary 2</h2>

<ul>
<li>At the read level:

<ul>
<li class='fragment'>Include bias in model for isoform abundances</li>
</ul></li>
<li class='fragment'>At the gene level:

<ul>
<li class='fragment'>Model on known covariates</li>
<li class='fragment'>Remove unknown batch differences with factor analysis</li>
<li class='fragment'>Remove any differences correlating with batch </li>
</ul></li>
</ul>

<p><br><br></p>

<p class='fragment'>count \(\sim\) \(\mathcal{L}\) (bias \(\cdot\) biology)</p>

    <aside class='notes'>
      
    </aside>
   </section>
</section>
<section class='' data-state='' id='slide-8'>
  <h2>Bias and batch</h2>
  <ul>
<li>Accurate measurements necessary for research</li>
<li>Way more important if used as markers</li>
</ul>

<p><br></p>

<p><img src="figure/mammaprint.jpg"></p>

</section>
    </div>
  </div>
</body>
  <script src="libraries/frameworks/revealjs/lib/js/head.min.js"></script>
  <script src="libraries/frameworks/revealjs/js/reveal.min.js"></script>
  <script>
  // Full list of configuration options available here:
  // https://github.com/hakimel/reveal.js#configuration
  Reveal.initialize({
    controls: true,
    progress: true,
    history: true,
    center: true,
    theme: Reveal.getQueryHash().theme || 'solarized', 
    transition: Reveal.getQueryHash().transition || 'linear', 
    dependencies: [
    // Cross-browser shim that fully implements classList -
    // https://github.com/eligrey/classList.js/
      { src: 'libraries/frameworks/revealjs/lib/js/classList.js', condition: function() { return !document.body.classList;}},
      // Zoom in and out with Alt+click
      { src: 'libraries/frameworks/revealjs/plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },
      // Speaker notes
      { src: 'libraries/frameworks/revealjs/plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } },
      // Remote control your reveal.js presentation using a touch device
      //{ src: 'libraries/frameworks/revealjs/plugin/remotes/remotes.js', async: true, condition: function() { return !!document.body.classList; } }
      ]
  });
  </script>  <!-- MathJax: Fall back to local if CDN offline but local image fonts are not supported (saves >100MB) -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [['$','$'], ['\\(','\\)']],
      processEscapes: true
    }
  });
</script>
<script type="text/javascript" src="http://cdn.mathjax.org/mathjax/2.0-latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
<!-- <script src="https://c328740.ssl.cf1.rackcdn.com/mathjax/2.0-latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script> -->
<script>window.MathJax || document.write('<script type="text/x-mathjax-config">MathJax.Hub.Config({"HTML-CSS":{imageFont:null}});<\/script><script src="libraries/widgets/mathjax/MathJax.js?config=TeX-AMS-MML_HTMLorMML"><\/script>')
</script>
<!-- LOAD HIGHLIGHTER JS FILES -->
<script src="libraries/highlighters/highlight.js/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<!-- DONE LOADING HIGHLIGHTER JS FILES -->
 

</html>